packages/libp2p/src/connection-manager/dial-queue.ts

```diff
@@ -457,6 +484,17 @@ export class DialQueue {
               pendingDial.status = 'active'
 
               let conn: Connection
+              let peer: Peer | undefined
+
+              try {
+                if (pendingDial.peerId != null) {
+                  peer = await this.peerStore.get(pendingDial.peerId)
```

Comment:
maybe this peerStore.get should happen in _updateAddressStatus and leave _updateAddressStatus signature alone?

I think it tracks more nicely to have the get+update in the "same place".
Also thats one less promise that has to be fulfilled before we get to dial.

Reply:
I take your point but when refactored that way a potential `ERR_NOT_FOUND` would be propagated in lieu of a consequent and perhaps more meaningful error such as a `NO_RESERVATION` error.